WordPress/Classes/ViewRelated/Stats/Insights/SiteStatsInsightsTableViewController.swift

```diff
@@ -412,16 +412,29 @@ extension SiteStatsInsightsTableViewController: SiteStatsInsightsDelegate {
         }
 
         if FeatureFlag.statsNewInsights.enabled {
-            let detailTableViewController = SiteStatsInsightsDetailsTableViewController()
-            detailTableViewController.configure(statSection: statSection, selectedDate: selectedDate)
-            navigationController?.pushViewController(detailTableViewController, animated: true)
+            switch statSection {
+            case .insightsViewsVisitors, .insightsFollowersWordPress, .insightsFollowersEmail, .insightsFollowerTotals:
```

Comment:
For a future improvement, I wonder if the detail view controller for a statSection could be provided by some extension on StatSection 🤔 If both `SiteStatsInsightsDetailsTableViewController` and `SiteStatsDetailTableViewController` conformed to a protocol that included the `configure` method, the resulting VC could simply be pushed onto the nav stack. Code here would become something like:

```
let detailViewController = statSection.detailViewController
detailViewController.configure(statSection: statSection, selectedData: selectedDate)
navigationController?.pushViewController(detailViewController, animated: true)
```

I guess this might not work if the detailVC differs depending on where it's being presented from. In which case we could just move this switch into a `detailViewController(for statSection)` method within this table VC.